common/v2/services/Store/BalanceService.tsx

```diff
@@ -176,18 +178,28 @@ export const getAccountsTokenBalances = (accounts: StoreAccount[], tokenContract
 
 // Unlock Token getBalance will return 0 if no valid unlock token is found for the address.
 // If there is an Unlock token found, it will return the id of the token.
-export const getAccountsUnlockVIPAddresses = async (accounts: StoreAccount[]) =>
-  getAccountsTokenBalance(accounts, MYCRYPTO_UNLOCK_CONTRACT_ADDRESS)
+export const getAccountMemberships = async (accounts: StoreAccount[]) =>
+  getAccountsTokenBalances(
+    accounts,
+    Object.values(MEMBERSHIP_CONFIG).map(c => c.contractAddress)
+  )
     .then(unlockStatusBalanceMap =>
-      Object.keys(unlockStatusBalanceMap).filter(address =>
-        unlockStatusBalanceMap[address].isGreaterThan(new BN(0))
+      R.fromPairs(
+        R.reject(
+          t => R.isEmpty(t[1]),
+          R.toPairs(
+            R.mergeAll(
+              Object.keys(unlockStatusBalanceMap).map(address => ({
+                [address]: Object.keys(unlockStatusBalanceMap[address])
+                  .filter(b => unlockStatusBalanceMap[address][b].isGreaterThan(new BN(0)))
+                  .map(b => MEMBERSHIP_CONTRACTS[b])
+              }))
+            )
+          )
+        )
```

Comment:
This whole thing is eye-cancer. Surely there has to be a better way to do this

Reply:
I rewrote this in another way which seems cleaner 😄